docs/sql-migration-guide.md

```diff
@@ -23,8 +23,11 @@ license: |
 {:toc}
 
 ## Upgrading from Spark SQL 3.0 to 3.1
+
   - Since Spark 3.1, grouping_id() returns long values. In Spark version 3.0 and earlier, this function returns int values. To restore the behavior before Spark 3.0, you can set `spark.sql.legacy.integerGroupingId` to `true`.
 
+  - Since Spark 3.1, the `formatted` query explain mode is used in the events (e.g., `SparkListenerSQLExecutionStart` and `SparkListenerSQLAdaptiveExecutionUpdate`) of SQL listeners. To restore the behavior before Spark 3.0, you can set `spark.sql.queryDescriptionModeInListeners` to `extended`.
```

Comment:
this doesn't break anything but just adds a new feature, do we really need a migration guide?

Reply:
Ur, yea. I added this because I thought user applications could listen to `SparkListenerSQLExecutionStart` and this PR changed the content in the event. But, I don't have a strong opinion on this, so I'll remove it.

Reply:
It's sort of behavior change and it worth mentioning the new configuration. I think it is fine to have the migration guide.

Reply:
We only add a migration guide if we change behavior BY DEFAULT.

We can mention it in release notes though, to promote this feature.